pypom/page.py

```diff
@@ -2,19 +2,63 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin
+
 from .view import WebView
 
 
 class Page(WebView):
 
-    _url = '{base_url}'
+    URL_TEMPLATE = ''
     """String representing a URL that will return this page.
 
     This string is formatted and can contain names of keyword arguments passed
-    during construcion of the page object. The ``{base_url}`` keyword argument
-    is always passed to the format operation.
+    during construction of the page object. The template should either assume
+    that the value of ``base_url`` will be prepended to the result of the
+    format, or should yield an absolute URL.
     """
 
+    @property
+    def target_url(self):
```

Comment:
I still feel like 'target' is misleading. It's more of a source than a target, at least when used by `open`. How do you feel about `canonical_url`?